PHPCompatibility/Sniffs/Syntax/InterpolatedStringDereferencingSniff.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2009-2020 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Sniffs\Syntax;
+
+use PHPCompatibility\Sniff;
+use PHPCSUtils\Utils\TextStrings;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Detect dereferencing of interpolated strings.
+ *
+ * PHP version 8.0
+ *
+ * @link https://wiki.php.net/rfc/variable_syntax_tweaks#interpolated_and_non-interpolated_strings
+ * @link https://www.php.net/manual/en/language.types.string.php#language.types.string.parsing
+ *
+ * @since 10.0
```

Comment:
```suggestion
 * @since 10.0.0
```
(here and elsewhere)

We use the semantic version in `@sìnce` tags for all code added since the switch to semantic versioning.

Background: in the "olden days", the versions of PHPCompatibility sort-of aligned with the PHP version number for which sniffs were being added.
The larger the codebase grew, the less manageable that became as it did not allow for us to make breaking changes, like supporting Composer out of the box (needed a directory layout change).
So, as of PHPCompatibility 8.0.0 semantic versioning is used.

See the changelog for more context: https://github.com/PHPCompatibility/PHPCompatibility/blob/develop/CHANGELOG.md